app/models/namespace.rb

```diff
@@ -111,6 +130,88 @@ def ransackable_attributes(_auth_object = nil)
     def ransackable_associations(_auth_object = nil)
       %w[]
     end
+
+    def subtract_from_metadata_summary_count(namespaces, metadata_summary, by_one)
+      return if metadata_summary.empty?
+
+      update_metadata_summary_counts(namespaces, metadata_summary, by_one: by_one, addition: false)
+    end
+
+    def add_to_metadata_summary_count(namespaces, metadata_summary, by_one)
+      return if metadata_summary.empty?
+
+      update_metadata_summary_counts(namespaces, metadata_summary, by_one: by_one, addition: true)
+    end
+
+    private
+
+    def update_metadata_summary_counts(namespaces, metadata_summary, by_one: false, addition: true) # rubocop:disable Metrics/AbcSize,Metrics/MethodLength
+      metadata_summary_update_sql = build_metadata_summary_update_sql(metadata_summary,
+                                                                      addition ? Arel::Nodes::Addition : Arel::Nodes::Subtraction,
+                                                                      by_one:)
+
+      entry_table = Arel::Table.new(
```

Comment:
This line is a good candidate for a function in the `PrivilegedNamespaceMetadataQuery` class